cubi_tk/sodar/check_remote.py

```diff
@@ -0,0 +1,441 @@
+            default=False,
+            action="store_true",
+            help="Flag to double check that md5 sums stored in local files do actually match their corresponding files",
+        )
+        parser.add_argument(
+            "--report-md5sums",
+            default=False,
+            action="store_true",
+            help="Flag to indicate if md5 sums should be included in file report",
+        )
+        parser.add_argument(
+            "--assay-uuid",
+            default=None,
+            type=str,
+            help="UUID from Assay to check. Used to specify target while dealing with multi-assay projects.",
+        )
+        parser.add_argument("project_uuid", type=str, help="UUID from Project to check.")
+
+    @classmethod
+    def run(
+        cls, args, _parser: argparse.ArgumentParser, _subparser: argparse.ArgumentParser
+    ) -> typing.Optional[int]:
+        """Entry point into the command."""
+        return cls(args).execute()
+
+    @staticmethod
+    def check_args(args):
+        """Called for checking arguments."""
+        res = 0
+
+        # If SODAR info not provided, fetch from user's toml file
+        toml_config = load_toml_config(args)
+        args.sodar_url = args.sodar_url or toml_config.get("global", {}).get("sodar_server_url")
+        args.sodar_api_token = args.sodar_api_token or toml_config.get("global", {}).get(
+            "sodar_api_token"
+        )
+
+        # Validate base path
+        if not os.path.exists(args.base_path):  # pragma: nocover
+            logger.error("Base path %s does not exist", args.base_path)
+            res = 1
+
+        return res
+
+    def execute(self) -> typing.Optional[int]:
+        """Execute the transfer."""
+        res = self.check_args(self.args)
+        if res:  # pragma: nocover
+            return res
+
+        logger.info("Starting cubi-tk sodar check-remote")
+        logger.info("  args: %s", self.args)
+
+        # Find all remote files (iRODS)
+        pseudo_args = SimpleNamespace(hash_scheme=DEFAULT_HASH_SCHEME)
+        irodscollector = RetrieveIrodsCollection(
+            pseudo_args,
+            self.args.sodar_url,
+            self.args.sodar_api_token,
+            self.args.assay_uuid,
+            self.args.project_uuid,
+        )
+
+        remote_files_dict = irodscollector.perform()
+        assay_path = irodscollector.get_assay_irods_path(self.args.assay_uuid)
+
+        # Find all local files with md5 sum
+        local_files_dict = FindLocalMD5Files(
+            base_path=self.args.base_path, recheck_md5=self.args.recheck_md5
+        ).run()
+
+        # Run checks
+        results = FileComparisonChecker(
+            remote_files_dict=remote_files_dict,
+            local_files_dict=local_files_dict,
+            filenames_only=self.args.filename_only,
+            irods_basepath=assay_path,
+        ).run()
+
+        if results:
+            logger.info("All done.")
+
+        return int(not results)
```

Comment:
I don't get it. Is there any circumstance that `FileComparisonChecker` won't return True?
Maybe this would be enough?
```python
        FileComparisonChecker(
            remote_files_dict=remote_files_dict,
            local_files_dict=local_files_dict,
            filenames_only=self.args.filename_only,
            irods_basepath=assay_path,
        ).run()

        logger.info("All done.")
        return 0
```

Reply:
Yes, that would equivalent - I can make the change